### PR TITLE
track cache usage

### PR DESCRIPTION
**Pull request by SrdjanStevanetic**

### CODE REVIEW

File: translation-tests/src/integration-test/java/eu/europeana/api/translation/tests/web/TranslationRestIT.java [hunk 1]
```diff
@@ -263,7 +263,7 @@ void translationPangeanicNoSrcMultipleLanguages() throws Exception {
             .accept(MediaType.APPLICATION_JSON)
             .contentType(MediaType.APPLICATION_JSON)
             .content(requestJson))
-        .andExpect(status().isOk())
+//        .andExpect(status().isOk())
```
Comment (gsergiu) [inline translation-tests/src/integration-test/java/eu/europeana/api/translation/tests/web/TranslationRestIT.java]: any reason why this was commented out?

File: translation-tests/src/integration-test/java/eu/europeana/api/translation/tests/BaseTranslationTest.java [hunk 2]
```diff
@@ -39,7 +39,7 @@
 public abstract class BaseTranslationTest extends IntegrationTestUtils {
 
   protected MockMvc mockMvc;
-  protected final static int redisPort=6370;
+  protected final static int redisPort=16370;
```
Comment (gsergiu) [inline translation-tests/src/integration-test/java/eu/europeana/api/translation/tests/BaseTranslationTest.java]: we should add redis test container to integration tests within the scope of this ticket (similar to mongo container is set api) https://testcontainers.com/modules/redis/

File: translation-web/src/main/java/eu/europeana/api/translation/web/service/CachedTranslationService.java [hunk 3]
```diff
@@ -51,18 +54,26 @@ public void translate(List<TranslationObj> translationObjs) throws TranslationEx
...
+        redisCacheService.store(toTranslate);  
+      }
     }
     
-    translationService.translate(toTranslate);
-    
-    if(isCachingEnabled()) {
-      //save result in the redis cache
-      redisCacheService.store(toTranslate);  
+    //logging the number of translated/cached lines and chars
```
Comment (gsergiu) [inline translation-web/src/main/java/eu/europeana/api/translation/web/service/CachedTranslationService.java]: please move the logging part to own method and use a java object to collect the metrics, e.g. TranslationCashingStats.

File: translation-web/src/main/java/eu/europeana/api/translation/web/service/CachedTranslationService.java [hunk 4]
```diff
@@ -51,18 +54,26 @@ public void translate(List<TranslationObj> translationObjs) throws TranslationEx
...
+    int numLinesCached=(int) translationObjs.stream().filter(el -> el.isRetrievedFromCache()).count();
+    int numCharsCached=translationObjs.stream().filter(el -> el.isRetrievedFromCache()).map(el -> el.getTranslation().length()).reduce(0, Integer::sum);
+    int numLinesTranslated=toTranslate.size();
+    int numCharsTranslated=toTranslate.stream().map(el -> el.getText().length()).reduce(0, Integer::sum);
+    if(logger.isInfoEnabled()) {
```
Comment (gsergiu) [inline translation-web/src/main/java/eu/europeana/api/translation/web/service/CachedTranslationService.java]: if the metrics are not logs, their computation is redundand, please update acordingly.

File: translation-web/src/main/java/eu/europeana/api/translation/web/service/CachedTranslationService.java [hunk 5]
```diff
@@ -51,18 +54,26 @@ public void translate(List<TranslationObj> translationObjs) throws TranslationEx
...
-    translationService.translate(toTranslate);
-    
-    if(isCachingEnabled()) {
-      //save result in the redis cache
-      redisCacheService.store(toTranslate);  
+    //logging the number of translated/cached lines and chars
+    int numLinesCached=(int) translationObjs.stream().filter(el -> el.isRetrievedFromCache()).count();
+    int numCharsCached=translationObjs.stream().filter(el -> el.isRetrievedFromCache()).map(el -> el.getTranslation().length()).reduce(0, Integer::sum);
+    int numLinesTranslated=toTranslate.size();
+    int numCharsTranslated=toTranslate.stream().map(el -> el.getText().length()).reduce(0, Integer::sum);
```
Comment (gsergiu) [inline translation-web/src/main/java/eu/europeana/api/translation/web/service/CachedTranslationService.java]: this is incorectly computed, as it must not include the texts discarded for translation (i.e. nonTranslatable & same language)

File: translation-web/src/main/java/eu/europeana/api/translation/web/service/CachedTranslationService.java [hunk 6]
```diff
@@ -51,18 +54,26 @@ public void translate(List<TranslationObj> translationObjs) throws TranslationEx
...
     }
     
-    translationService.translate(toTranslate);
-    
-    if(isCachingEnabled()) {
-      //save result in the redis cache
-      redisCacheService.store(toTranslate);  
+    //logging the number of translated/cached lines and chars
+    int numLinesCached=(int) translationObjs.stream().filter(el -> el.isRetrievedFromCache()).count();
+    int numCharsCached=translationObjs.stream().filter(el -> el.isRetrievedFromCache()).map(el -> el.getTranslation().length()).reduce(0, Integer::sum);
```
Comment (gsergiu) [inline translation-web/src/main/java/eu/europeana/api/translation/web/service/CachedTranslationService.java]: for performance reasons it is better to use a regular for loop, in which both the lines and the chars are computed.

Comment (gsergiu) [reply translation-web/src/main/java/eu/europeana/api/translation/web/service/CachedTranslationService.java]: the computation of the lines and chars server by the translation service must be computed as well, 
see last row in the comment: https://europeana.atlassian.net/browse/EA-3969?focusedCommentId=133237

**Review by gsergiu — CHANGES_REQUESTED** [review]

File: translation-tests/src/integration-test/java/eu/europeana/api/translation/tests/web/TranslationRestIT.java (hunk 1 above)
Comment (SrdjanStevanetic) [reply translation-tests/src/integration-test/java/eu/europeana/api/translation/tests/web/TranslationRestIT.java]: Removed, it works fine.

**Pull request merged**